unitTests/test_basicio.cpp

```diff
@@ -19,6 +19,42 @@ TEST(MemIo, seek_out_of_bounds)
     // The problem is that MemIo::seek() does not check that the new offset is
     // in bounds.
     byte tmp[16];
-    io.seek(0x10000000, BasicIo::beg);
+    ASSERT_EQ(io.seek(0x10000000, BasicIo::beg), 1);
+    ASSERT_TRUE(io.eof());
+
+    // The seek was invalid, so the offset didn't change and this read still works.
+    ASSERT_EQ(io.read(tmp, sizeof(tmp)), sizeof(tmp));
+
+    // Reset the file.
```

Comment:
[suggestion] It seems that every time you are resetting, your aim is to make a slightly different test/check. I would propose to create different tests for each case, and try to give meaningful names to each of them.

If that's not the case or you prefer to not make more changes here let me know.